float64 totalUT
string assigned_edge